### PR TITLE
Add per-host shard metrics if configured

### PR DESCRIPTION
**Pull request by grubernaut**
Allows users to get per-host shard metrics from `shardConnPoolStats`.

Since this adds additional measurement from the `mongodb` plugin, it is
configured via the telegraf configuration file.

Also adds unit test and documentation for the changeset.

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by danielnelson** [comment]
Since it doesn't take another request to get this data, I don't think we need an option to control it.  Users can remove it using the normal [measurement filtering](https://github.com/influxdata/telegraf/blob/master/docs/CONFIGURATION.md#measurement-filtering) options.

**Comment by grubernaut** [comment]
@danielnelson awesome, makes sense. Sorry for late reply, was on vacation, will fix and push an update by EOD.

**Comment by grubernaut** [comment]
AppVeyor seems to be failing from a non-related issue: 
```
ERROR restoring some imports:
-   Error cloning repo at C:\gopath\src\github.com\stretchr\objx, exit status 128
-   Error cloning repo at C:\gopath\src\github.com\StackExchange\wmi, exit status 128
-   Error cloning repo at C:\gopath\src\github.com\streadway\amqp, exit status 128
-   Error cloning repo at C:\gopath\src\github.com\stretchr\testify, exit status 128
-   Error cloning repo at C:\gopath\src\github.com\tidwall\gjson, exit status 128
-   Error cloning repo at C:\gopath\src\github.com\yuin\gopher-lua, exit status 128
-   Error cloning repo at C:\gopath\src\github.com\vjeantet\grok, exit status 128
-   Error cloning repo at C:\gopath\src\github.com\tidwall\match, exit status 128
-   Error cloning repo at C:\gopath\src\github.com\wvanbergen\kafka, exit status 128
-   Error cloning repo at C:\gopath\src\github.com\wvanbergen\kazoo-go, exit status 128
Error cloning repo at C:\gopath\src\github.com\zensqlmonitor\go-mssqldb, exit status 128
make: *** [deps] Error 1
```

**Comment by danielnelson** [comment]
Looks like a network issue, hopefully it will clear up soon and I'll try to run the build again periodically.

**Comment by danielnelson** [comment]
Appveyor team helped me get the tests running again.

**Comment by grubernaut** [comment]
@danielnelson, fixed merge conflicts, any target release to be able to include this? Thanks again! :tada:

File: plugins/inputs/mongodb/mongodb_data.go [hunk 1]
```diff
@@ -179,4 +203,14 @@ func (d *MongodbData) flush(acc telegraf.Accumulator) {
 		)
 		db.Fields = make(map[string]interface{})
 	}
+	for _, host := range d.ShardHostData {
+		d.Tags["shard_host_name"] = host.Name
```
Comment (danielnelson) [inline plugins/inputs/mongodb/mongodb_data.go]: Should this be just `hostname` to match existing tags in other measurements?

File: plugins/inputs/mongodb/mongodb_data.go [hunk 2]
```diff
@@ -179,4 +203,14 @@ func (d *MongodbData) flush(acc telegraf.Accumulator) {
...
+		acc.AddFields(
+			"mongodb_shard_host_stats",
```
Comment (danielnelson) [inline plugins/inputs/mongodb/mongodb_data.go]: Would this be better as `mongodb_shard_stats` since we have `mongodb_db_stats`?

**Comment by grubernaut** [comment]
@danielnelson updated! thanks for taking a look!

**Comment by danielnelson** [comment]
Merged for 1.7, thanks!

**Comment by grubernaut** [comment]
Awesome, thanks!

**Pull request merged**